build: cannot find -ltinfo

```
$ make build
GOPATH set to /home/mjibson
Running make with -j12
GEN pkg/cli/zcgo_flags.go
GEN pkg/server/status/zcgo_flags.go
GEN pkg/storage/engine/zcgo_flags.go
GEN pkg/ccl/storageccl/engineccl/zcgo_flags.go
make[1]: Nothing to be done for 'build_lib_static'.
[ 62%] Built target snappy
[100%] Built target snappy-unittest
[100%] Built target roach
[ 85%] Built target roach
[100%] Built target roachccl
[100%] Built target libprotobuf
[  0%] Built target build_version
[100%] Built target rocksdb
go build -i -o ./cockroach -v  -tags ' make x86_64_linux_gnu' -ldflags ' -X github.com/cockroachdb/cockroach/pkg/build.typ=development -X "github.com/cockroachdb/cockroach/pkg/build.tag=v1.2-alpha.20170901-695-gcb3503fc4" -X "github.com/cockroachdb/cockroach/pkg/build.utcTime=2017/10/09 01:09:56" -X "github.com/cockroachdb/cockroach/pkg/build.rev=cb3503fc4ba9737e7a02a274dc763018a1c70102" -X "github.com/cockroachdb/cockroach/pkg/build.baseBranch=v1.2-alpha.20170901"' .
github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix
# github.com/cockroachdb/cockroach/vendor/github.com/knz/go-libedit/unix
/usr/bin/ld: cannot find -ltinfo
collect2: error: ld returned 1 exit status
Makefile:191: recipe for target 'build' failed
make: *** [build] Error 2
```

The recent go-libedit changes aren't building on my system (debian).

I had to `sudo apt-get install libtinfo-dev` to get this to work. It is my opinion that we should not include C code that requires 3rd-party packages to be installed because of stuff like this.

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -31,6 +31,9 @@ new engineers.
      - A C++ compiler that supports C++11. Note that GCC prior to 6.0 doesn't
        work due to https://gcc.gnu.org/bugzilla/show_bug.cgi?id=48891
      - The standard C/C++ development headers on your system.
+     - On GNU/Linux, the terminfo development libraries, which may be
+       part of a ncurses development package (e.g. `libtinfo-dev` on
+       Debian/Ubuntu, but `ncurses-devel` on CentOS).
      - A Go environment with a recent 64-bit version of the toolchain. Note that
        the Makefile enforces the specific version required, as it is updated
        frequently.

